File: Data Structures & Algorithms/valid-binary-search-tree/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool helper(TreeNode* node, int minimum, int maximum){
        if(!node) return true;
        if(node->val <= minimum || node->val >= maximum) return false;

        return helper(node->left, minimum, node->val) && helper(node->right, node->val, maximum);
    }

    bool isValidBST(TreeNode* root) {
        return helper(root, INT_MIN, INT_MAX);
    }
};

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        while(not root):
            return result

        q = deque([root])
        while(len(q) > 0):
            temp = []
            length = len(q)
            for i in range(length):
                node = q.popleft()
                temp.append(node.val)

                if(node.left):
                    q.append(node.left)
                if(node.right):
                    q.append(node.right)

            result.append(temp[-1])
        
        return result